File: src/main/java/land/leets/domain/portfolio/domain/repository/PortfolioRepository.java
package land.leets.domain.portfolio.domain.repository;

import land.leets.domain.portfolio.domain.Portfolio;
import land.leets.domain.portfolio.domain.ProjectScope;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface PortfolioRepository extends JpaRepository<Portfolio, Long> {
    List<Portfolio> findAllByGeneration(Long generation);

    List<Portfolio> findAllByGenerationAndScope(Long generation, ProjectScope scope);

    List<Portfolio> findAllByScopeOrderByGenerationDesc(ProjectScope scope);

    Portfolio findByPortfolioId(Long portfolioId);
}

File: src/main/kotlin/land/leets/domain/portfolio/domain/repository/PortfolioRepository.kt
package land.leets.domain.portfolio.domain.repository

import land.leets.domain.portfolio.domain.Portfolio
import land.leets.domain.portfolio.type.ProjectScope
import org.springframework.data.jpa.repository.JpaRepository

interface PortfolioRepository : JpaRepository<Portfolio, Long> {
    fun findAllByGenerationAndScope(generation: Long, scope: ProjectScope): List<Portfolio>

    fun findAllByScopeOrderByGenerationDesc(scope: ProjectScope): List<Portfolio>
}
